clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10159,13 +10163,24 @@ void Cgeist::ConstructJob(Compilation &C, const JobAction &JA,
 
   TCArgs.AddAllArgValues(CmdArgs, options::OPT_Xcgeist);
 
+  const InputInfo *HostModule = nullptr;
   for (const InputInfo &I : Inputs) {
-    if (I.isFilename())
+    if (I.getType() == types::TY_MLIR_IR)
+      HostModule = &I;
+    else if (I.isFilename())
       CmdArgs.push_back(I.getFilename());
     else
       I.getInputArg().renderAsInput(TCArgs, CmdArgs);
   }
 
+  if (HostModule) {
+    // If the output is MLIR, this will avoid dropping host code
+    if (!TCArgs.hasArg(options::OPT_fsycl_device_only))
+      CmdArgs.push_back("-sycl-device-only=false");
```

Comment:
Why do we need to pass `-sycl-device-only=false` explicitly rather than passing nothing?

Reply:
I changed the code so that this option only works with opaque pointers.